File: week2/problemset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Declare function protoypes
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for the text
    string text = get_string("Text: ");

    // Retrieve the number of letters, words and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate the L and S values
    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;

    // Calculate and round the index value
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    // Print the grade depending on the index value.
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// Function to count the number of letters in the text string
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

// Function to count the number of words in the text string
int count_words(string text)
{
    int words = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }

    words++;

    return words;
}

// Function to count the number of sentences in the text string
int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        switch (text[i])
        {
            case '.':
                sentences++;
                break;

            case '!':
                sentences++;
                break;

            case '?':
                sentences++;
                break;

            default:
                break;
        }
    }

    return sentences;
}

File: week6/problemset6/sentimental-readability/readability.py
from cs50 import get_string

# Define a function to count the number of letters in a string.
def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters

# Define a function to count the number of words in a string.
def count_words(text):
    words = 0
    for i in range(len(text)):
        if text[i] == " ":
            words += 1
    words += 1
    return words

# Define a function to count the number of sentences in a string.
def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] in [".", "!", "?"]:
            sentences += 1
    return sentences

# Get the text string from the user.
text = get_string("Text: ")

# Count the number of letters, words, and sentences in the text string.
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

# Calculate the L and S values.
L = letters / words * 100
S = sentences / words * 100

# Calculate and round the index value.
index = 0.0588 * L - 0.296 * S - 15.8
grade = int(round(index))

# Print the grade depending on the index value.
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade", grade)
